core/os/device/deviceinfo/cc/android/query.cpp

```diff
@@ -501,7 +500,7 @@ const char* gpuName() { return ""; }
 
 const char* gpuVendor() { return ""; }
 
-const char* instanceName() { return gContext.mSerial.c_str(); }
+const char* instanceName() { return gContext.mHardware.c_str(); }
```

Comment:
We already return the hardware name below. Why would we have two functions returning the same thing? Can you please fix what calls these functions instead?

Reply:
It looks like we have a duplicate in device.instance: device.instance.name == device.instance.configuration.hardware.name , at least in android. See:

https://github.com/google/agi/blob/77648d1557836fcf549c4238e65fb1e35c1d296f/core/os/android/adb/device.go#L245

Should we get rid of the device.instance.name field entirely?

Reply:
Ok, sorry, I was confusing things. It is fine to return the hardware name here.